lib/post_voting/comment_review_queue.rb

```diff
@@ -0,0 +1,151 @@
+# frozen_string_literal: true
+module PostVoting
+  class CommentReviewQueue
+    def flag_comment(comment, guardian, flag_type_id, opts = {})
+      result = { success: false, errors: [] }
+      is_notify_user_type = ReviewableScore.types.slice(:notify_user).values.include?(flag_type_id)
+      is_notify_moderators_type =
+        ReviewableScore.types.slice(:notify_moderators).values.include?(flag_type_id)
+      is_notify_type = is_notify_user_type || is_notify_moderators_type
+
+      guardian.ensure_can_flag_post_voting_comment!(comment)
+      guardian.ensure_can_flag_post_voting_comment_as!(comment, flag_type_id, opts)
+
+      existing_reviewable = Reviewable.includes(:reviewable_scores).find_by(target: comment)
+
+      if !can_flag_again?(existing_reviewable, comment, guardian.user, flag_type_id)
+        result[:errors] << I18n.t("post_voting.reviewables.comment_already_handled")
+        return result
+      end
+
+      payload = { comment_cooked: comment.cooked }
+
+      if opts[:comment].present? && is_notify_type
+        creator = companion_pm_creator(comment, guardian.user, flag_type_id, opts)
+        post = creator.create
+
+        if creator.errors.present?
+          creator.errors.full_messages.each { |msg| result[:errors] << msg }
+          return result
+        end
+      end
+
+      queued_for_review = !!ActiveRecord::Type::Boolean.new.deserialize(opts[:queue_for_review])
+
+      if !is_notify_user_type
+        reviewable =
+          ReviewablePostVotingComment.needs_review!(
+            created_by: guardian.user,
+            target: comment,
+            reviewable_by_moderator: true,
+            potential_spam: flag_type_id == ReviewableScore.types[:spam],
+            topic: comment.post.topic,
+            payload: payload,
+          )
+        reviewable.update(target_created_by: comment.user)
+        score =
+          reviewable.add_score(
+            guardian.user,
+            flag_type_id,
+            meta_topic_id: post&.topic_id,
+            take_action: opts[:take_action],
+            reason: queued_for_review ? "post_voting_comment_queued_by_staff" : nil,
+            force_review: queued_for_review,
+          )
+
+        if opts[:take_action]
+          reviewable.perform(guardian.user, :agree_and_delete)
+        else
+          enforce_auto_silence_threshold(reviewable)
+        end
+      end
+
+      result.tap do |r|
+        r[:success] = true
+        r[:reviewable] = reviewable if !is_notify_user_type
+      end
+    end
+
+    private
+
+    def enforce_auto_silence_threshold(reviewable)
+      auto_silence_duration = SiteSetting.chat_auto_silence_from_flags_duration
+      return if auto_silence_duration.zero?
+      return if reviewable.score <= Chat::ReviewableMessage.score_to_silence_user
+
+      user = reviewable.target_created_by
+      return if user.admin?
+      return unless user
+      return if user.silenced?
```

Comment:
Maybe it should be something like: 

```suggestion
      return unless user
      return if user.admin?
      return if user.silenced?
```